libs/infinity_emb/infinity_emb/inference/select_model.py

```diff
@@ -84,9 +88,11 @@ def select_model(
         )
         logger.info(log_msg)
         # now warm up with max_token, max batch size
-        loaded_engine.warmup(batch_size=engine_args.batch_size, n_tokens=512)
+        loaded_engine.warmup(
+            batch_size=engine_args.batch_size, n_tokens=loaded_engine.max_length
+        )
         emb_per_sec, _, log_msg = loaded_engine.warmup(
-            batch_size=engine_args.batch_size, n_tokens=512
+            batch_size=engine_args.batch_size, n_tokens=loaded_engine.max_length
         )
```

Comment:
@wirthual 512 means 512 or less, whatever the tokenizer constrains it to. 

- Do you think adding max_length adds more complexity. 
- Also, for some architectures (jina-8192) this will add large overhead.

Reply:
Thats a good point. I can use 509 as max_length, that seems to work well.